exporter/elasticsearchexporter/model.go

```diff
@@ -163,6 +169,125 @@ func (m *encodeModel) encodeLogECSMode(resource pcommon.Resource, record plog.Lo
 	return document
 }
 
+func (m *encodeModel) encodeMetrics(resource pcommon.Resource, metrics pmetric.MetricSlice, _ pcommon.InstrumentationScope) ([][]byte, error) {
+	var baseDoc objmodel.Document
+
+	baseDoc.AddAttributes("", resource.Attributes())
+
+	// Put all metrics that have the same attributes and timestamp in one document.
+	docs := map[uint32]*objmodel.Document{}
+	for i := 0; i < metrics.Len(); i++ {
+		metric := metrics.At(i)
+
+		var dps pmetric.NumberDataPointSlice
+
+		// Only Gauge and Sum metric types are supported at the moment.
+		switch metric.Type() {
+		case pmetric.MetricTypeGauge:
+			dps = metric.Gauge().DataPoints()
+		case pmetric.MetricTypeSum:
+			dps = metric.Sum().DataPoints()
+		}
+
+		for j := 0; j < dps.Len(); j++ {
+			dp := dps.At(j)
+
+			hash := metricHash(dp.Timestamp(), dp.Attributes())
+			doc, docExists := docs[hash]
+			if !docExists {
+				doc = baseDoc.Clone()
```

Comment:
IIUC `Clone` is not a deep clone, and dedup is potentially going to mutate the field. I feel more comfortable with a deep clone. In any case, can you add a test where in resource attributes and dp attributes, there are duplicate fields, as well as renamed fields (see Dedup logic), and ensure that the output is still expected?